### PR TITLE
Update to the new proposed Visual Viewport API

### PR DESCRIPTION
**Pull request by darktears**
Proposed API:
https://wicg.github.io/visual-viewport/#dom-visualviewport-segments

The new API now supports multiple folds so update the code accordingly.

### CODE REVIEW

**Comment by darktears** [comment]
@kenchris PTAL

File: docs/global.html [hunk 1]
```diff
@@ -449,13 +360,36 @@ <h5 class="subsection-title">Properties:</h5>
...
+            <td class="type">
+            
+                
+<span class="param-type">number</span>
+
+
+            
+            </td>
+
+
```
Comment (kenchris) [inline docs/global.html:375]: what is up with all these newlines?

File: docs/index.js.html [hunk 2]
```diff
@@ -122,52 +132,69 @@ <h1 class="page-title">Source: index.js</h1>
   }
 
   getSegments() {
-    if (this.screenSpanning === "none") {
+    if (this.verticalViewportSegments === 1 &amp;&amp; this.horizontalViewportSegments === 1) {
```
Comment (kenchris) [inline docs/index.js.html:135]: `&amp;&amp;` that doesn't look right

File: docs/index.js.html [hunk 3]
```diff
@@ -122,52 +132,69 @@ <h1 class="page-title">Source: index.js</h1>
...
+      let topOffset = 0;
       const width = window.innerWidth;
-      return [
-        { top: 0, left: 0, width, height: screenCenter - this.foldSize / 2 },
-        { top: screenCenter - this.foldSize / 2, height: this.foldSize, left: 0, width },
-        { top: screenCenter + this.foldSize / 2, left: 0, width, height: screenCenter - this.foldSize / 2 }
-      ];
+      const height = availableHeight / this.horizontalViewportSegments - this.foldSize *
+        numberOfFolds / this.horizontalViewportSegments;
+      for (let i = 0; i &lt; this.horizontalViewportSegments + numberOfFolds; ++i) {
```
Comment (kenchris) [inline docs/index.js.html:149]: `&lt;` something wrong here as well

File: docs/index.js.html [hunk 4]
```diff
@@ -122,52 +132,69 @@ <h1 class="page-title">Source: index.js</h1>
...
+        if (i % 2 === 0)
+          segments[i] = { top: topOffset, left: 0, width: width, height: height };
+        else
```
Comment (kenchris) [inline docs/index.js.html:152]: big mistake here :) no {} and last line is then outside of the if sentence... please always use {}

File: docs/index.js.html [hunk 5]
```diff
@@ -122,52 +132,69 @@ <h1 class="page-title">Source: index.js</h1>
...
       const height = window.innerHeight;
-      return [
-        { top: 0, left: 0, width, height},
-        { top: 0, height, left: width, width: this.foldSize },
-        { top: 0, left: window.innerWidth / 2 + this.foldSize / 2, width, height }
-      ];
+      let leftOffset = 0;
+
+      for (let i = 0; i &lt; this.verticalViewportSegments + numberOfFolds; ++i) {
+        if (i % 2 === 0)
```
Comment (kenchris) [inline docs/index.js.html:166]: Please add {}

File: docs/index.js.html [hunk 6]
```diff
@@ -122,52 +132,69 @@ <h1 class="page-title">Source: index.js</h1>
...
+        numberOfFolds / this.verticalViewportSegments;
       const height = window.innerHeight;
-      return [
-        { top: 0, left: 0, width, height},
-        { top: 0, height, left: width, width: this.foldSize },
-        { top: 0, left: window.innerWidth / 2 + this.foldSize / 2, width, height }
-      ];
+      let leftOffset = 0;
+
+      for (let i = 0; i &lt; this.verticalViewportSegments + numberOfFolds; ++i) {
```
Comment (kenchris) [inline docs/index.js.html:165]: `&lt;` again

**Comment by darktears** [comment]
@kenchris you're reviewing the docs/ directory. This is a direct output of jsdocs, untouched :).

**Pull request merged**